### ISSUE TITLE
[A11Y] Permissions page checkboxes are not accessible

### ISSUE THREAD

**Issue by quicksketch** [backdrop/backdrop-issues#4987]
**Description of the bug**

This problem was raised by @MustafaDogan78 in issue #4985.

The checkboxes on the permissions page (`admin/config/people/permissions`) are not accessible via a screen reader. Instead of each role and permission being read by the screen reader, @MustafaDogan78 reports that the checkboxes must be counted to associate which checkbox matches each role.

The original report (as run through Google Translate):

> I don't know English. I'm building a dialect with translation. It's a Problem, but there's no other way. There is another problem with accessibility. Setting permissions. Problem reading roles first, then checkboxes. When using Drupal, the cue box and the roles would be read simultaneously. I figure out what role to allow by counting the tick boxes.

**Steps To Reproduce**

To reproduce the behavior:
1. Enable a screen reader such as NVDA on Windows or VoiceOver on Mac.
2. Go to 'admin/config/people/permissions'
3. Use the tab key to focus on individual checkboxes that grant a permission
4. Listen to the voice over.

**Actual behavior**

The voice over says the word "Checkbox" but does not say what the checkbox does.

**Expected behavior**

The voice over should announce the name of the role for each checkbox.

**Additional information**

@MustafaDogan78 reports that Drupal 7 has a more desireable behavior that we may compare against. However, I think we intentionally reworked this page to avoid "element-invisible" labels because it made using the browser search difficult, because the permission name was repeated in the hidden text for every role, resulting in a confusing and tedious search functionality.

With improved accessibility options since Drupal 7, we should investigate if using an `aria-` attribute can provide an alternative that is both accessible to screen readers but does not cause a usability problem.

**Comment by quicksketch** [comment]
In Linux, focusing a checkbox also announces the _group_ of the permissions, rather than the permission name itself. For example, focusing the "Anonymous" role of the "Administer custom blocks" permission, announces "Block Checkbox, not checked". Completely unhelpful and actually inaccurate.

**Comment by quicksketch** [comment]
I made an attempt to fix this problem with a PR at https://github.com/backdrop/backdrop/pull/3558.

It uses the the [`aria-label`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-label_attribute) attribute to give each checkbox a label friendly to screen readers but does not create hidden markup (such as using the `element-invisible` class) on the page.

The label I chose is "[role] permission: [permission name]".

With the Linux screen reader Orca, it announces: "Authenticated permission: Administer custom blocks. Checkbox unchecked."

I did some minor refactoring of variable names in the code to match our coding standards.

**Comment by MustafaDogan78** [comment]
Windows has a built-in screen reader, but it is not preferred. The blind don't like him. However, those who want to try can turn it on and off with the Ctrl + start + enter keys.
I wouldn't recommend setting up JAWS. It may distort the video card settings.
I'll leave the NVDA screen reader link for Windows here. NVDA is open source. I've prepared a portable for you. It will not be installed on your computer. Use insert + q keys to close. To turn write mode on or off, insert + space. NVDA screen reader link.
www.mustafadogan.net.tr/nvda.exe

**Comment by quicksketch** [comment]
Thank you @MustafaDogan78! It has been several years since I lasted used a screen reader on Windows and I had forgot that NVDA is the new standard. I have updated the issue summary to suggest using NVDA.

**Comment by indigoxela** [comment]
Sorry, I'm a bit late here.

Could using the attribute "aria-labelledby" simplify things regarding duplications? Currently the same text gets added to every checkbox in a row with "aria-label". Maybe `aria-labelledby="id-of-row-title"` provides the same functionality without duplicating the text? But maybe that causes problems with the additional description.
